src/events/integration_hub.py

```diff
@@ -47,15 +47,21 @@ def _remove_resources(self, _):
     def _on_integration_hub_pebble_ready(self, _):
         """Handle on Pebble ready event."""
         self.integration_hub.update(
-            self.context.s3, self.context.pushgateway, self.context.hub_configurations
+            self.context.s3,
+            self.context.azure_storage,
+            self.context.pushgateway,
+            self.context.hub_configurations,
```

Comment:
**thought, out-of-scope** I believe we need to improve the design here. I also found it a bit annoying that whenever we add a new relation/something, we need to modify tons of handlers. Maybe we could just feed the context in the constructor, and be explicit what we want to override in the update (to allow for setting None of some options). Anyhow, this is out of the scope of the current PR, but I thought that it was worth mentioning, such as you guys can also have a thought